### PR TITLE
feat(boost): add FindBoost

### PR DESCRIPTION
**Pull request by sfoster1**
This adds FindBoost.cmake, which will download and use boost by using the incredibly awesome https://github.com/Orphis/boost-cmake which evades most of the normal problems of using boost (like, it's gigantic and takes forever to build and needs to be preinstalled, oh god) by providing cmake scaffolding to not only integrate boost into your build, but, like, integrate boost into your build with cmake semantics so only the components you link in actually get built, and they get built at build time, and it's just really nice. Will help us make simulators. You can check out how it works at https://github.com/Opentrons/opentrons-modules/pull/175

### CODE REVIEW

File: boost-cmake/CMakeLists.txt [hunk 1]
```diff
@@ -0,0 +1,104 @@
...
+  target_compile_definitions(Boost::boost INTERFACE
+    $<$<CONFIG:Release>:BOOST_DISABLE_ASSERT>
+    BOOST_SYSTEM_NO_DEPRECATED
+    BOOST_THREAD_VERSION=4
+    BOOST_THREAD_USES_CHRONO
+    BOOST_THREAD_PROVIDES_EXECUTORS
+  )
+endif()
+
+if(USE_ANDROID)
```
Comment (Laura-Danielle) [inline boost-cmake/CMakeLists.txt:101]: Same comment as above.

File: boost-cmake/CMakeLists.txt [hunk 2]
```diff
@@ -0,0 +1,104 @@
...
+message(STATUS "Boost found: ${BOOST_VERSION} ${BOOST_SOURCE}")
+
+include(StandaloneBuild)
+include(PlatformDetect)
+include(AddBoostLib)
+include(AddBoostTest)
+
+set_property(GLOBAL PROPERTY USE_FOLDERS TRUE)
+
+if(USE_ANDROID)
```
Comment (Laura-Danielle) [inline boost-cmake/CMakeLists.txt:36]: Do we need this? Is someone ever going to run this on android?

Comment (sfoster1) [reply boost-cmake/CMakeLists.txt:36]: everything in `boost-cmake` is subtreed in from https://github.com/Orphis/boost-cmake and we shouldn't change it locally

**Review by Laura-Danielle — APPROVED** [review]
lgtm

**Pull request merged**